plugin/storage/es/spanstore/reader.go

```diff
@@ -238,19 +238,40 @@ func (s *SpanReader) FindTraces(traceQuery *spanstore.TraceQueryParameters) ([]*
 	if err != nil {
 		return nil, err
 	}
-	var retMe []*model.Trace
-	for _, traceID := range uniqueTraceIDs {
-		if len(retMe) >= traceQuery.NumTraces {
-			break
-		}
-		trace, err := s.readTrace(traceID, traceQuery)
+	return s.multiRead(uniqueTraceIDs, traceQuery)
+}
+
+func (s *SpanReader) multiRead(traceIDs []string, traceQuery *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
```

Comment:
looks like we can remove the readTrace function entirely given it's a subset of this function